src/remoted/secure.c

```diff
@@ -622,24 +650,30 @@ STATIC void HandleSecureMessage(const message_t *message, int *wdb_sock) {
             _close_sock(&keys, message->sock);
         }
 
+        if (sock_idle >= 0) {
+            _close_sock(&keys, sock_idle);
+        }
+
         rem_inc_recv_unknown();
         return;
     }
 
+    /* Recieved valid message timestamp and protocol updated. */
+    keys.keyentries[agentid]->rcvd = current_ts;
+    keys.keyentries[agentid]->net_protocol = protocol;
```

Comment:
Thinking about setting the protocol, I think we should move this to were it was before because everything related to the connection is updated with the keep-alive messages (also update the comment from https://github.com/wazuh/wazuh/pull/18085/files#diff-edf09aaabb1573d2d5c0737e6fc3e0ab4ba124b093888c53a377dce3612cc9ddR661).

Reply:
Done!